sound/soc/sof/intel/hda.c

```diff
@@ -437,10 +437,10 @@ int hda_dsp_probe(struct snd_sof_dev *sdev)
 	 * use position update IPC if either it is forced
 	 * or we don't have other choice
 	 */
-#if IS_ENABLED(CONFIG_SND_SOC_SOF_DEBUG_FORCE_IPC_POSITION)
-	hdev->no_ipc_position = 0;
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_DEBUG_DISABLE_NOWAKEUP)
+	hdev->ipc_nowakeup = 0;
 #else
-	hdev->no_ipc_position = sof_ops(sdev)->pcm_pointer ? 1 : 0;
+	hdev->ipc_nowakeup = sof_ops(sdev)->pcm_pointer ? 1 : 0;
```

Comment:
Not related to this patch directly, but I found this a bit confusing. Here ipc_nowakeup is set based on pcm_pointer() availability, but then e.g. in hda_dsp_pcm_pointer() 'ipc_nowakeup' is checked again and there are two codepaths. There seems to be duplicate logic in either of the places. What do you think?

Reply:
I agree this feature is not clear and some part is confusing. I understand your point.
But as per your question, ipc_nowakeup is 0 when SND_SOC_SOF_DEBUG_DISABLE_NOWAKEUP is enabled. ipc_nowakeup in hda_dsp_pcm_pointer() has to be checked again.